specs/SBOM_SPEC.md

```diff
@@ -0,0 +1,109 @@
+# Cosign SBOM Specifications
+
+This document aims to describe how `cosign` attaches SBOM (Software Bill of Materials) documents to containers.
+
+The goal is to specify the behavior well enough to promote other implementations and enable interoperability.
+SBOMs attached with `cosign` should be retrievable in other tools, and vice-versa.
+
+This document focuses on the layout of an SBOM within an [OCI Image Manifest V1](https://github.com/opencontainers/image-spec/blob/master/manifest.md) object.
+
+This document does not prescribe any specific SBOM format.
+Multiple formats can be used, and the `mediaType` property should describe the format of a particular SBOM document.
+
+Multiple SBOMs may be "attached" to one image.
+Each SBOM may refer to the entire image, or to a specific part of that image.
+Exactly what the SBOM refers to is called the "scope" of that SBOM.
+
+SBOMs stored in an OCI registry are generally assumed to refer to other objects stored in that same registry.
+This document does not specify how these "links" are created.
+A naming convention or the [in-progress OCI `references` API](https://github.com/opencontainers/image-spec/issues/827) are viable options.
+
+This document does not specify how clients should behave When multiple SBOMs are present for an image.
+Clients may list all the SBOMs, or may provide tooling to filter based on SBOM type or scope.
+
+## Overall Layout
+
+An SBOM object is represented an [OCI Image Manifest V1](https://github.com/opencontainers/image-spec/blob/master/manifest.md).
+
+Each individual SBOM is represented as a `layer`, using a standard `descriptor`.
+The `layers` list is ordered, but no order is assumed or important for the SBOM documents.
+
+Here is an example manifest containing one SBOM, in the [SPDX](https://spdx.org) format:
+
+```json
+{
+  "schemaVersion": 2,
+  "config": {
+    "mediaType": "application/vnd.oci.image.config.v1+json",
+    "size": 233,
+    "digest": "sha256:83bd5fb5b39f65f28e50a86d48fa79c07880befc292d92eebdc18531054b070c"
+  },
+  "layers": [
+    {
+      "mediaType": "text/spdx",
+      "size": 246,
+      "digest": "sha256:ed3ad03d3b87843b5419d7dce9d50a3e0f45554b2ba93bf378611cae6b450cff",
+    }
+  ]
+}
+```
+
+Multiple SBOMs may be attached, using multiple formats.
+This example shows two SBOMs, one in the SPDX format and one in the CycloneDX format:
+
+```json
+{
+  "schemaVersion": 2,
+  "config": {
+    "mediaType": "application/vnd.oci.image.config.v1+json",
+    "size": 233,
+    "digest": "sha256:83bd5fb5b39f65f28e50a86d48fa79c07880befc292d92eebdc18531054b070c"
+  },
+  "layers": [
```

Comment:
+1 on having multiple boms of different formats that can point to different layers -- would be super helpful on the buildpacks side for https://github.com/buildpacks/rfcs/pull/166